src/App.vue

```diff
@@ -1,85 +1,29 @@
 <script setup lang="ts">
+import { Bars3Icon } from '@heroicons/vue/24/outline'
 import { RouterLink, RouterView } from 'vue-router'
-import HelloWorld from './components/HelloWorld.vue'
-</script>
 
-<template>
-  <header>
-    <img alt="Vue logo" class="logo" src="@/assets/logo.svg" width="125" height="125" />
+import { useModal } from '@/components/modal/composables/useModal'
+import ModalWrapper from '@/components/modal/ModalWrapper.vue'
+import SideDrawer from '@/layouts/SideDrawer.vue'
+import SideNavigation from '@/layouts/SideNavigation.vue'
 
-    <div class="wrapper">
-      <HelloWorld msg="You did it!" />
+const { shouldShowModal, closeModal, handleOpenModal } = useModal()
+</script>
 
-      <nav>
-        <RouterLink to="/">Home</RouterLink>
-        <RouterLink to="/about">About</RouterLink>
-      </nav>
-    </div>
+<template>
+  <header class="flex gap-2 h-12 w-full items-center pl-2 shadow">
+    <button class="flex items-center md:hidden" @click="handleOpenModal">
+      <Bars3Icon class="size-8" />
+    </button>
+    <RouterLink to="/">
+      <img alt="Vue logo" class="size-8" src="@/assets/logo.svg" />
+    </RouterLink>
   </header>
-
-  <RouterView />
+  <ModalWrapper v-if="shouldShowModal" @close-modal="closeModal">
+    <SideDrawer />
+  </ModalWrapper>
+  <div class="flex">
+    <SideNavigation class="hidden md:block" />
```

Comment:
とりあえずは以下のようにすることで `gap-2` が有効になりました
メディアクエリで画面幅が 768 以上のときに `display: block;` が当てられていて、これと flex とが競合していたようです
```suggestion
    <SideNavigation class="hidden" />
```